### PR TITLE
build: add bom module

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: mqtt-bom/build.gradle [hunk 1]
```diff
@@ -0,0 +1,10 @@
+plugins {
+    id "io.micronaut.build.internal.bom"
+}
+dependencies {
+    constraints {
+        api("org.eclipse.paho:org.eclipse.paho.client.mqttv3:$pahov3Version")
```
Comment (timyates) [inline mqtt-bom/build.gradle:6]: This is an `implementation` class in the `mqttv3` project, not an `api` one 🤔

**Review by timyates — CHANGES_REQUESTED** [review]

File: mqtt-bom/build.gradle (hunk 1 above)
Comment (sdelamo) [reply mqtt-bom/build.gradle:6]: do you think we should not expose it in the BOM? or do you think we should change it to be `api` in `mqttv3`. I think the latter probably makes more sense.

Comment (timyates) [reply mqtt-bom/build.gradle:6]: I agree it should probably be the latter

Comment (sdelamo) [reply mqtt-bom/build.gradle:6]: Resolved via https://github.com/micronaut-projects/micronaut-mqtt/pull/128/commits/eaae750a75b094339d8508954ea752543ec67e61

**Pull request merged**